src/utils.cpp

```diff
@@ -482,5 +495,241 @@ nlohmann::json getParsedJson(const std::string& pathToJson)
         throw std::runtime_error("Failed to parse JSON file");
     }
 }
+
+void executePostFailAction(const nlohmann::json& i_parsedConfigJson,
+                           const std::string& i_vpdFilePath,
+                           const std::string& i_flagToProcess)
+{
+    if (i_parsedConfigJson.empty() || i_vpdFilePath.empty() ||
+        i_flagToProcess.empty())
+    {
+        throw std::runtime_error(
+            "Invalid parameters. Abort processing for post fail action");
+    }
+
+    if (!(i_parsedConfigJson["frus"][i_vpdFilePath].at(0).contains(
+            "postActionFail")))
+    {
+        throw std::runtime_error(
+            "postActionFail flag missing in config JSON. Abort processing");
+    }
+
+    for (const auto& l_tags : (i_parsedConfigJson["frus"][i_vpdFilePath].at(
+             0))["postActionFail"][i_flagToProcess]
+                                  .items())
+    {
+        auto itrToFunction = funcionMap.find(l_tags.key());
+        if (itrToFunction != funcionMap.end())
+        {
+            if (!itrToFunction->second(i_parsedConfigJson, i_vpdFilePath,
+                                       "postActionFail", i_flagToProcess))
+            {
+                break;
+            }
+        }
+    }
+
+    return;
+}
+
+bool processSystemCmdTag(const nlohmann::json& i_parsedConfigJson,
+                         const std::string& i_vpdFilePath,
+                         const std::string& i_baseAction,
+                         const std::string& i_flagToProcess)
+{
+    if (i_vpdFilePath.empty() || i_parsedConfigJson.empty() ||
+        i_baseAction.empty() || i_flagToProcess.empty())
+    {
+        logging::logMessage("Invalid parameter");
+        return false;
+    }
+
+    if ((i_parsedConfigJson["frus"][i_vpdFilePath].at(
+             0)[i_baseAction][i_flagToProcess]["systemCmd"])
+            .contains("cmd"))
+    {
+        const std::string& l_bindCommand =
+            i_parsedConfigJson["frus"][i_vpdFilePath].at(
+                0)[i_baseAction][i_flagToProcess]["systemCmd"]["cmd"];
+
+        logging::logMessage("Bind command = " + l_bindCommand);
+        utils::executeCmd(l_bindCommand);
```

Comment:
do we just let the exception directly go out ?

Reply:
Added handling.